contracts/consensus/Consensus.sol

```diff
@@ -91,29 +103,28 @@ contract Consensus is MasterCopyNonUpgradable, CoreStatusEnum, ConsensusI {
     /** Coinbase split per mille */
     uint256 public coinbaseSplitPerMille;
 
-    /** Block hash of heads of Metablockchains */
-    mapping(bytes20 /* chainId */ => bytes32 /* MetablockHash */) public metablockHeaderTips;
+    /** Mapping of a metablock number to a metablock (per chain).  */
+    mapping(bytes20 => mapping(uint256 => Metablock)) public metablockchains;
 
-    /** Core statuses */
-    mapping(address /* core */ => CoreStatus /* coreStatus */) public coreStatuses;
+    /** Metablocks' heights per chain. */
+    mapping(bytes20 => uint256) public metablockTips;
 
     /** Assigned core for a given chainId */
     mapping(bytes20 /* chainId */ => address /* core */) public assignments;
 
-    /** Precommitts from cores for metablockchains. */
-    mapping(address /* core */ => Precommit) public precommits;
+    /** Core statuses. */
+    mapping(address /* core */ => CoreStatus /* coreStatus */) public coreStatuses;
+
+    /** Linked-list of committees. */
+    mapping(address => address) public committees;
```

Comment:
```
mapping(bytes32 /* metachainId */ => CommitteeI) public committees;
```

Reply:
that way, there is less state build-up on `Consensus`; when a new metablock is precommitted; this mapping can be overwritten.

In return the pruning of `Committees` can be done from the committee itself; it can check whether it can be selfdestructed by checking open balances, and whether it has reported the decision / latest, if the metablockchain has moved on to another blockheight

Reply:
We can create a follow up ticket for this. What are your thoughts?